src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,153 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String EDGE_LOGIN = "EdgeLo";
+    private static final String INVALID_LOGIN = "invL";
+    private static final String VALID_PASSWORD = "password";
+    private static final String EDGE_PASSWORD = "EdgePs";
+    private static final String INVALID_PASSWORD = "invP";
+    private static final int VALID_USER_AGE = 20;
+    private static final int USER_EDGE_AGE = 18;
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+    private static User nullLoginPasswordUser;
+    private static User invalidAgeUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        nullLoginPasswordUser = new User();
+        invalidAgeUser = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        nullLoginPasswordUser.setAge(VALID_USER_AGE);
+        nullLoginPasswordUser.setLogin(null);
+        nullLoginPasswordUser.setPassword(null);
+        invalidAgeUser.setLogin(VALID_LOGIN);
+        invalidAgeUser.setPassword(VALID_PASSWORD);
+    }
```

Comment:
To simplify your code, you can initialize several default users in BeforeEach method and modify their fields with needed parameters in different tests